ui-react/packages/atlasmap/src/Atlasmap/utils/document.ts

```diff
@@ -95,3 +100,155 @@ export async function importInstanceSchema(
+    .toPromise()
+    .then((classPath: string) => {
+      cfg.initCfg.classPath = classPath;
+      cfg.documentService
+        .fetchDocument(docdef, cfg.initCfg.classPath)
+        .toPromise()
+        .then(async (doc: DocumentDefinition) => {
+          // No fields indicate the user is attempting to enable a custom
+          // field action class.  Remove the document from the panel since
+          // it has no fields.
+          if (doc.fields.length === 0) {
+            // Make any custom field actions active.
+            await cfg.fieldActionService
+              .fetchFieldActions()
+              .catch((error: any) => {
+                cfg.errorService.addError(
+                  new ErrorInfo({
+                    message: error,
+                    level: ErrorLevel.ERROR,
+                    scope: ErrorScope.APPLICATION,
+                    type: ErrorType.INTERNAL,
+                  }),
+                );
+              });
+
+            if (doc.isSource) {
+              DataMapperUtil.removeItemFromArray(doc, cfg.sourceDocs);
+            } else {
+              DataMapperUtil.removeItemFromArray(doc, cfg.targetDocs);
+            }
+          }
+          await cfg.mappingService.notifyMappingUpdated();
+          await cfg.fileService.exportMappingsCatalog("");
+        })
+        .catch((error: any) => {
+          if (error.status === 0) {
+            cfg.errorService.addError(
+              new ErrorInfo({
+                message: `Unable to fetch the Java class document '${docdef.name}' from the runtime service.`,
+                level: ErrorLevel.ERROR,
+                scope: ErrorScope.APPLICATION,
+                type: ErrorType.INTERNAL,
+                object: error,
+              }),
+            );
+          } else {
+            cfg.errorService.addError(
+              new ErrorInfo({
+                message: `Could not load the Java class document '${docdef.id}'`,
+                level: ErrorLevel.ERROR,
+                scope: ErrorScope.APPLICATION,
+                type: ErrorType.INTERNAL,
+                object: error,
+              }),
+            );
+          }
+        });
+    })
+    .catch((error: any) => {
+      if (error.status === 0) {
+        cfg.errorService.addError(
+          new ErrorInfo({
+            message:
+              "Fatal network error: Could not connect to AtlasMap design runtime service.",
+            level: ErrorLevel.ERROR,
+            scope: ErrorScope.APPLICATION,
+            type: ErrorType.INTERNAL,
+            object: error,
+          }),
+        );
+      } else {
+        cfg.errorService.addError(
+          new ErrorInfo({
+            message: "Could not load the Java class path.",
+            level: ErrorLevel.ERROR,
+            scope: ErrorScope.APPLICATION,
+            type: ErrorType.INTERNAL,
+            object: error,
+          }),
+        );
+      }
+    });
+}
+
+interface ISelectValues1 {
```

Comment:
I’m curious, why the 1 in the name?

Reply:
Sorry - I should have been clearer - what I really want is for a FormSelect to list all of the class names defined by the user's imported JARs,  That will take more work on the server side - for now we're doing what the angular Ui did and having the user type the full class.  This interface is used by getCustomClassCandidates() which is currently not called.  It will be when I implement the select pulldown later on.

Reply:
corrected - I took out my partial fix - will keep it on a dev branch